code/game/machinery/doors/firedoor.dm

```diff
@@ -78,17 +82,153 @@
 	. = ..()
 	INVOKE_ASYNC(src, .proc/latetoggle)
 
-/obj/machinery/door/firedoor/attack_hand(mob/user, list/modifiers)
+/obj/machinery/door/firedoor/attack_hand(mob/living/user, list/modifiers)
 	. = ..()
 	if(.)
 		return
 	if(operating || !density)
 		return
-	user.changeNext_move(CLICK_CD_MELEE)
+	if(user.combat_mode)
+		user.changeNext_move(CLICK_CD_MELEE)
 
-	user.visible_message("<span class='notice'>[user] bangs on \the [src].</span>", \
-		"<span class='notice'>You bang on \the [src].</span>")
-	playsound(loc, 'sound/effects/glassknock.ogg', 10, FALSE, frequency = 32000)
+		user.visible_message("<span class='notice'>[user] bangs on \the [src].</span>", \
+			"<span class='notice'>You bang on \the [src].</span>")
+		playsound(loc, 'sound/effects/glassknock.ogg', 10, FALSE, frequency = 32000)
+	else
+		if(user.do_afters)//you get one firelock to open, better pick the right one
+			return
+		number_of_people_trying_to_open++
+
+		//dont want them to be moved by pressure differences when theyre holding onto the firelock with all their strength
+		user.move_resist = MOVE_FORCE_VERY_STRONG
+
+		//starting stats in order to write logs that provide enough data to make kickass fucking graphs later
+		var/starting_health = user.health
+		var/starting_time = REALTIMEOFDAY
+
+		var/datum/gas_mixture/users_air = user.return_air()
+		var/user_loc_starting_temperature = users_air.temperature
+
+		///we dont want people with only one hand to have the same visible_message as people with two hands
+		var/hand_string = "hands"
+		if(iscarbon(user))
+			var/mob/living/carbon/user_as_carbon = user
+			var/obj/item/bodypart/right_arm = user_as_carbon.get_bodypart(BODY_ZONE_R_ARM)
+			var/obj/item/bodypart/left_arm = user_as_carbon.get_bodypart(BODY_ZONE_L_ARM)
+
+			if(!right_arm || right_arm.bodypart_disabled)//this is assuming that you have at least one hand since... this proc is called attack_hand
+				hand_string = "hand"
+			if(!left_arm || left_arm.bodypart_disabled)
+				hand_string = "hand"
+
+		//we want to acknowledge that several people are trying to open the same firelock to encourage them to do it
+		//also to signal to people doing it by themselves that its faster with help
+		if(number_of_people_trying_to_open == 1)
+			user.visible_message("<span class='notice'>[user] tries to open \the [src] with their [hand_string], struggling greatly to open the heavy door by themselves.</span>", \
+				"<span class='notice'> You try with all your strength to pry open \the [src] with your [hand_string], barely moving it!</span>")
+		else if (number_of_people_trying_to_open == 2)
+			user.visible_message("<span class='notice'>[user.name] joins another in trying to open \the [src] with their [hand_string].</span>", \
+				"<span class='notice'> You join another in trying to pry open \the [src] with your [hand_string]!</span>")
```

Comment:
```suggestion
				"<span class='notice'>You join another in trying to pry open \the [src] with your [hand_string]!</span>")
```

Reply:
Also, can this really not just say the name of the other person?

Reply:
now says the name of the other person when youre the second to join the do_after